backend/src/api/image/delete_image.rs

```diff
@@ -0,0 +1,40 @@
+use anyhow::Context;
+use axum::{extract::Path, Extension, Json};
+use rusqlite::params;
+use std::fs;
+use tracing::*;
+
+use std::sync::Arc;
+
+use crate::api::auth::Authorize;
+use crate::api::error::Error;
+use crate::AppState;
+
+pub(super) async fn delete(
+    Path(image_key): Path<String>,
+    Authorize(user): Authorize,
+    Extension(state): Extension<Arc<AppState>>,
+) -> Result<Json<()>, Error> {
+    let mut image_path = state.data_path.clone();
+    image_path.push(&image_key);
```

Comment:
Is it possible to escape the path here? eg. `DELETE /api/images/../../root`

Reply:
That's a really good point which I did not consider but nothing should actually happen in that case since we check if a user owns the image and in effect if the image key is valid before this path is actually used.